### PR TITLE
Feature/profile page

### LINKED ISSUE #116: Profile Page
**__Task Description__**
Create the profile page

**__Technical Considerations__**
This page will require functionality from the backend that will not be ready at the time of creation.

**__Additional Considerations__**
Any supplemental information including unresolved questions, links to external resources, screenshots, etc.

### PR DESCRIPTION
**Pull request by moody2times**
# Profile Page

- Created the profile page and necessary components
- Wrote unit and integration tests
- Styled the profile page
- Add a reducer function to dispatch on delete user
- Created another modal for image because it may cause merge conflict to modify existing modal, which cannot be reused

closes #116

### CODE REVIEW

File: client/src/store/features/users/usersSlice.js [hunk 1]
```diff
@@ -105,11 +106,37 @@ export function unFriend(userToUpdate, id, token) {
...
+    const friendToRemove = currentFriends.find(
+      (friend) => !updatedFriends.some((f) => f.id === friend.id),
+    )
     dispatch(removeFriend(friendToRemove.id))
   }
 }
 
+// todo: consider using the createAsyncThunk middleware
+// async thunk to send delete user request
+export function deleteProfile(userId) {
```
Comment (KhadidjaArezki) [inline client/src/store/features/users/usersSlice.js]: Just a reminder here that the delete and put requests to the db requires a valid token

File: client/src/styles/Profile-styles/EditProfile.module.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.Edit__heading {
+  border-bottom: 1px solid #c4c4c4;
+  margin-bottom: 5rem;
+}
+
+.Edit__button {
+  background-color: var(--clr-royal-blue);
+  box-shadow: 0 2rem 0 rgba(#000, 0.4);
```
Comment (KhadidjaArezki) [inline client/src/styles/Profile-styles/EditProfile.module.css]: Browser devtools show 'invalid property value' flag for this rule

**Review by KhadidjaArezki — CHANGES_REQUESTED** [review]
That is an impressive amount of commits. Kudos to you.
It looks great, but I will request a few changes before merging:
1. Edit picture button is way off on some mobile screens
Capture

2. Heading font is too large on mobile
CaptureHeadings

3. Cancel button on the edit page should redirect to the profile page.

File: client/src/store/features/users/usersSlice.js (hunk 1 above)
Comment (moody2times) [reply client/src/store/features/users/usersSlice.js]: You're right, thanks

File: client/src/styles/Profile-styles/EditProfile.module.css (hunk 2 above)
Comment (moody2times) [reply client/src/styles/Profile-styles/EditProfile.module.css]: Will fix it, nice catch.

**Comment by moody2times** [comment]
Yeah, I didn't write media queries for every single screen size because I was just following the sizes in the Figma board. I will add the cancel button functionality but I did mention that I did not add all button functionality just yet because this task was already taking too long. There are other smaller tasks that I could quickly finish, so I had to leave some functionality for now.
But, I'm going to fix the ones you took note of.

File: client/src/styles/Profile-styles/EditProfile.module.css (hunk 2 above)
Comment (moody2times) [reply client/src/styles/Profile-styles/EditProfile.module.css]: I said `rgba` but passed in `hex` values to the `box-shadow`, no wonder. I am so used to writing it like that using `sass` that I failed to pay attention that this is plain `css`.

**Comment by moody2times** [comment]
> That is an impressive amount of commits. Kudos to you. It looks great, but I will request a few changes before merging:
> 
> 1. Edit picture button is way off on some mobile screens
>    Capture
> 2. Heading font is too large on mobile
>    CaptureHeadings
> 3. Cancel button on the edit page should redirect to the profile page.

Implemented these changes. It is ready for review.

**Comment by KhadidjaArezki** [comment]
It looks great. Fantastic work.

**Review by KhadidjaArezki — APPROVED** [review]

**Pull request merged**